### ISSUE TITLE
Invariant image on Apple M1

### ISSUE THREAD

**Issue by solidpixel** [ARM-software/astc-encoder#273]
Testing for `toktx` is flagging some invariance between x64 and NEON builds. 

https://github.com/KhronosGroup/KTX-Software/blob/master/tests/srcimages/FlightHelmet_baseColor.png

```
astcenc -cl ../KTX-Software/tests/srcimages/FlightHelmet_baseColor.png helmet_intel_6x6.astc 6x6 -medium
```

**Comment by solidpixel** [comment]
Testing locally:

* `astcenc-none` and `astcenc-neon` differ
* `astcenc-none` and `astcenc-sse4.1` differ
* `astcenc-sse4.1` and `astcenc-neon` are the same

**Comment by solidpixel** [comment]
Apple silicon:

```
9ae9faec862593860af1923429a09f81 out-none.astc
811f8bec88f53a3fa824a082cb82fedb out-neon.astc
811f8bec88f53a3fa824a082cb82fedb out-sse2.astc
811f8bec88f53a3fa824a082cb82fedb out-sse4.1.astc
```

x86_64 machine:

```
811f8bec88f53a3fa824a082cb82fedb  out-none.astc
811f8bec88f53a3fa824a082cb82fedb  out-sse2.astc
811f8bec88f53a3fa824a082cb82fedb  out-avx2.astc
```

So looks to be something with the "none" build on Apple silicon (not sure if this is native or running in Rosetta - need to check)

**Comment by abbaswasim** [comment]
> Problem 2 - toktx should really be picking up the SSE2 version; NONE is a lot slower ...

**Comment by solidpixel** [comment]
In which case I don't quite understand where the invariance is coming from in the `toktx` testing, as all of the SIMD flavors are invariant with respect to each other in my testing here. 

Can you confirm the CMake configure line you are using in a repro outside of the `toktx` build system?

**Comment by wasimabbas-arm** [comment]
Here are the commands I have been using:

**Comment by solidpixel** [comment]
Minified test case - a single 6x6 block.

![in-min](https://user-images.githubusercontent.com/23016617/125689987-25c0ed4b-645b-42d8-a18f-ae1989344441.png)

**Comment by solidpixel** [comment]
Bug in NONE vector library. 

* Fixed on main in 43f9b6c
* Cherry-picked to 2.x branch in 1d13484
